core/src/test/java/io/dekorate/utils/GitTest.java

```diff
@@ -1,74 +1,126 @@
- * 
+ * <p>
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- * 
- * 
- * 
- * 
-**/
+ **/
 
 package io.dekorate.utils;
 
 import java.io.File;
 import java.net.URL;
 import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Test;
-import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
 
-class GitTest {
-
-  static String GIT_CONFIG = "git-simple/git/config";
-  static URL CONFIG_URL = GitTest.class.getClassLoader().getResource(GIT_CONFIG);
-  static File CONFIG_FILE = Urls.toFile(CONFIG_URL);
-  static Path DOT_GIT = CONFIG_FILE.toPath().getParent().getParent().resolve(Git.DOT_GIT);
-  static Path ROOT = CONFIG_FILE.toPath().getParent().getParent();
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 
+class GitTest {
+  
+  private static final ClassLoader loader = GitTest.class.getClassLoader();
+  private static final String CONFIG_SUFFIX = "/git/config";
+  private static final String GIT_SIMPLE = "git-simple";
+  private static final String GIT_SSH = "git-ssh";
+  private static final String GIT_GITLAB = "git-gitlab";
+  private static final Map<String, Path> configurationNameToConfigRoot = new HashMap<>(7);
+  
   @BeforeAll
-  public static void setup() {
-    CONFIG_FILE.toPath().getParent().toFile().renameTo(DOT_GIT.toFile());
+  static void setup() {
+    setup(GIT_SIMPLE);
+    setup(GIT_SSH);
+    setup(GIT_GITLAB);
   }
-
-  @Test
-  public void shouldDetectRoot() {
-    Path root = Git.getRoot(CONFIG_FILE.toPath());
-    assertEquals(ROOT, root);
+  
+  private static Path getDotGit(File configFile) {
+    return configFile.toPath().getParent().getParent().resolve(Git.DOT_GIT);
   }
-
-  @Test
-  public void shouldGetUrl() throws Exception {
-    Optional<String> repoUrl = Git.getRemoteUrl(ROOT, Git.ORIGIN);
+  
+  private static Path getRoot(File configFile) {
+    return configFile.toPath().getParent().getParent();
+  }
+  
+  public static void setup(String gitConfig) {
```

Comment:
This and the methods below don't have to be `public`. Maybe they should be package-private like the rest?